kolibri/core/device/models.py

```diff
@@ -93,6 +100,10 @@ def save(self, *args, **kwargs):
         super(DeviceSettings, self).save(*args, **kwargs)
         cache.set(DEVICE_SETTINGS_CACHE_KEY, self, 600)
 
+    def delete(self, *args, **kwargs):
+        super(DeviceSettings, self).delete(*args, **kwargs)
```

Comment:
Parsing the obfuscated Manager code, it seems that perhaps this method isn't necessary? It looks like it should be utilizing the `delete` method on the QuerySet. Otherwise, it seems like this should return the result of `super.delete`.

Reply:
Gotcha - can try removing this method and seeing if the tests still pass.

Reply:
Add the manager test, it passed. Removed the model method, and it failed.